tools/c7n_oci/c7n_oci/query.py

```diff
@@ -54,9 +54,27 @@ def get_resources(self, query):
     def _get_resources_for_list_of_compartment_ids(self, compartment_ids, list_func_ref):
         resources = []
         for compartment_id in compartment_ids:
-            results = self._get_resources_with_compartment_and_params(compartment_id, list_func_ref)
-            for result in results:
-                resources.append(oci.util.to_dict(result))
+            cache_key = self.get_cache_key_for_compartment(compartment_id)
+            cached_resources = self.manager._cache.get(cache_key)
+            if cached_resources:
+                self.manager.log.debug(
+                    "Using cached %s: %d"
+                    % (
+                        "%s.%s" % (self.__class__.__module__, self.__class__.__name__),
```

Comment:
s/self.manager.type is more appropriate here as it reflects the name the user is using in a policy.